### DISCUSSION TITLE
StreamDesk+

### DISCUSSION THREAD

**Discussion by I8NHJ** [Zayik/CommandSender#19] [General]
Thank you for your useful plug-in. Do you have any plan for extending it to use knobs and touch strip as provided by the + model?
Best,
Max

**Comment by Zayik** [comment]
Hey Max, 
I'm interested to know what support you'd image the knobs and strips having. 
Unfortunately, I don't have the + model so it'd be hard for me to test any new functionality as well. 

I appreciate your interest in the tool and glad it's working nicely for you.

**Comment by I8NHJ** [comment]
Hi David.

Your plug-in has been gaining some attention within the Ham Radio community
for controlling specific applications.

One notable example is a widely used software called N1MM+, which listens
for UDP packets to activate certain functions.

N1MM Logger Plus – Free Software for Phone, CW & Digital
<https://n1mmwp.hamdocs.com/>

This software automatically controls the connected radio.

There are some intriguing possibilities for integrating the knobs with this
software, including functions like adjusting Morse code speed, changing
frequency, and more.

Let me know if you're interested in exploring this further, and I’ll be
happy to assist you in any way I can.

On Tue, Dec 17, 2024 at 1:48 PM David Anderson ***@***.***>
wrote:

> Hey Max,
> I'm interested to know what support you'd image the knobs and strips
> having.
> Unfortunately, I don't have the + model so it'd be hard for me to test any
> new functionality as well.
>
> I appreciate your interest in the tool and glad it's working nicely for
> you.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Zayik/CommandSender/discussions/19#discussioncomment-11597669>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AF5QSL7G5GXMW257M4CAVOD2GB52LAVCNFSM6AAAAABTZEYHZGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNJZG43DMOI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
----------------------
73 de Max, N5NHJ (I8NHJ)

**Reply by vandenkm** [reply]
I have just discovered this plug-in from the N1MM Logger+ docs and I've been having a field day preparing for Winter Field Day :)

Really the need is to have a place to be able to create some "variables" in the plug-in.  Such as RFGain that can hold a numeric value, perhaps with lower/upper limits.  Then a turn of the knob to the left or the right can adjust the value, which then can be referenced in the "command" to be sent.  This could also be done by button actions...so a pair of buttons could perform an up/down operation on that value and then send the command.  This is NOT an N1MM valid command, but an example:

Var MonitorLevel [0..100] = 0

button 1 Command Pressed:  <CmdSndrVar  name="MonitorLevel" operation="+" value="5" /><Cmd name="{CAT1ASC ML0%MonitorLevel%;}" />

it quickly gets tricky...especially if a specific formatting of the value would be required (which it would for this particular use case).

Perhaps instead of constructing a whole new variable storage (and clean-up) concept the "current state" could itself be the "value" which could be incremented or decremented by some action (button press or knob turn). You'd just have a lot more states (in this case, up to 101)

**Reply by vandenkm** [reply]
Of course I started on the Elgato Marketplace and followed links over here completely forgetting this is github and the code is "right here".

I've got decades of C# coding experience but zero Stream Deck SDK or plug-in experience.  But it seems like a second plug-in action could be created to use a "Value" integer property instead of "state".  For my use case, I'm using the same communication type, IP Address and Port for all of the states. The only difference in my "command" is the hard-coded string version of the numeric value.

Perhaps it's time to clone, do a little experimenting and submit a pull request... ;)

**Reply by Zayik** [reply]
I'm glad you are all finding interest. And I do apologize for the delayed response. Christmas time hit and I completely missed the message. 

In terms of how to process the knob values, it'd make sense to make a separate action so we can properly make use of the knob functionality. I suspect that libraries I'm using have added knob support to some extent I could make use of. 

This was a little side project I worked on ages ago, and it intrigues me to know that other tools docs reference my tool :)

I think for me, the biggest hurdle for thinking of implementing knob functionality is just missing the hardware, and I can't justify spending $200 for the stream deck with knob functionality to properly test anything I create.

**Comment by I8NHJ** [comment]
Hi David,
If you’re willing to maintain the plugin and further develop it to support the SD+, I’m confident we can find people interested and willing to contribute financially for the required hardware.
Just let me know.

**Comment by Zayik** [comment]
That sounds good to me. I've charged $50/hour on past projects with a max
hour cap. I'd be fine for just cost of hardware. I'd create a new action
that operates on sending commands as knobs are turned.   This would include
user feedback, bug fixes, etc to ensure you all are happy with how it
works. I'm pretty flexible since this is a fun side project of mine. Let me
know what you think.

If anyone is interested in this tool being expanded to include specific new
features, I do commission as well. Feel free to reach out to me if
interested.

**Comment by I8NHJ** [comment]
I’ve been in touch with the developer of the "Commander" StreamDeck plugin to discuss his availability to support and extend the plugin for the StreamDeck+ (the model with knobs and touch strip).
David is enthusiastic about supporting the plugin and the ham radio community, but he doesn’t have a Stream Deck+, which is essential for development.
The unit costs around $200, and I believe this group can easily raise the funds to purchase one for David. If you’d like to be part of this initiative, please reach out to me via direct message
73, Max N5NHJ

**Comment by K3TN** [comment]
I've  just bought a Stream Deck+ and am trying to use Command Sender, can't get it work. Is it not supported yet?

Thanks and 73 John K3TN
